libsolidity/inlineasm/AsmParser.cpp

```diff
@@ -227,6 +229,18 @@ assembly::VariableDeclaration Parser::parseVariableDeclaration()
 	return varDecl;
 }
 
+assembly::ForLoop Parser:parseForLoop()
+{
+	ForLoop forLoop = createWithLocation<ForLoop>();
+	expectToken(Token::For);
+	forLoop.preCondition = parseBlock();
+	forLoop.expression = parseExpression();
+	forLoop.postCondition = parseBlock();
```

Comment:
The post condition? I think the original definition of having everything as a block, bar the conditional is the best so far.